sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/PushTokenApiRequestTest.kt

```diff
@@ -22,6 +23,18 @@ internal class PushTokenApiRequestTest : BaseRequestTest() {
         .setPhoneNumber(PHONE)
         .setExternalId(EXTERNAL_ID)
 
+    @Test
+    fun `Equality operator`() {
+        val aRequest = PushTokenApiRequest(PUSH_TOKEN, stubProfile)
+        val bRequest = PushTokenApiRequest(PUSH_TOKEN, stubProfile)
+        assertEquals(aRequest, bRequest)
+
+        // At this time, decoding from JSON does not preserve equality check for push token requests
+        // But at least it should not crash to compare them
+        val bRequestDecoded = KlaviyoApiRequestDecoder.fromJson(bRequest.toJson())
+        assertNotEquals(aRequest, bRequestDecoded)
```

Comment:
What's different between these?

Reply:
The cause of the bug was that we use a different constructor when decoding this class from JSON. So the first is just create 2 objects, test their equality. The second is encode and decode it, and re-test equality

Reply:
Previously, testing equality failed on a decoded json object because the lateinit var was not initialized. With this patch, equality check should not crash, though expected behavior is that it will not be equal anymore, because the payload body has been mutated

Reply:
Exactly, this was my failing-test repro.